hs_core/views/resource_folder_hierarchy.py

```diff
@@ -158,7 +158,7 @@ def data_store_structure(request):
             if not main_extension:
                 # accept any extension
                 main_extension = ""
-            if main_extension.endswith(f.extension):
+            if main_extension.endswith(f.extension) and f.extension:
```

Comment:
Should the order be changed in the if statement, i.e., change it to ```if f.extension and main_extension.endswith(f.extension)```?

Reply:
Good catch. Fixed.